telegram/request/_httpxrequest.py

```diff
@@ -122,15 +138,22 @@ def __init__(
             raise ValueError("`http_version` must be either '1.1', '2.0' or '2'.")
 
         http1 = http_version == "1.1"
-
+        http_kwargs = {"http1": http1, "http2": not http1}
+        transport = (
+            httpx.AsyncHTTPTransport(
+                socket_options=socket_options,
+            )
+            if socket_options
+            else None
+        )
         # See https://github.com/python-telegram-bot/python-telegram-bot/pull/3542
         # for why we need to use `dict()` here.
-        self._client_kwargs = dict(  # pylint: disable=use-dict-literal  # noqa: C408
+        self._client_kwargs = dict(  # pylint: disable=use-dict-literal
```

Comment:
tbh, this `disable=use-dict-literal` makes no sense. probably was a different problem.

not that it really matters. just an academic discussion.